lib/service.js

```diff
@@ -5259,46 +4940,43 @@
-         *              .pivot(function(pivotErr, pivot) {
-         *                  console.log("Pivot search is:", pivot.search);
-         *              });
-         *      });
-         *
-         * @param {Function} callback A function to call when done getting the pivot: `(err, pivot)`.
+         *      let dataModels = await service.dataModels().fetch();
+         *      let searches = dataModels.item("internal_audit_logs").objectByName("searches");
+         *      let pivotSpec = searches.createPivotSpecification();
+         *      // Use of the fluent API
+         *      let pivot = await pivotSpec.addRowSplit("user", "Executing user")
+         *          .addRangeColumnSplit("exec_time", {start: 0, end: 12, step: 5, limit: 4})
+         *          .addCellValue("search", "Search Query", "values")
+         *          .pivot();
+         *      console.log("Pivot search is:", pivot.search);
+         * 
+         * @param {Number} response_timeout A timeout period for aborting a request in milisecs (0 means no timeout).
          *
          * @method splunkjs.Service.PivotSpecification
          */
-        pivot: function(callback) {
-            var svc = this.dataModelObject.dataModel.service;
+        pivot: function(response_timeout) {
+            let svc = this.dataModelObject.dataModel.service;
 
-            var args = {
+            let args = {
                 pivot_json: JSON.stringify(this.toJsonObject())
             };
 
             if (!utils.isUndefined(this.accelerationNamespace)) {
                 args.namespace = this.accelerationNamespace;
             }
             
-            return svc.get(Paths.pivot + "/" + encodeURIComponent(this.dataModelObject.dataModel.name), args, function(err, response) {
-                if (err) {
-                    callback(new Error(err.data.messages[0].text), response);
-                    return;
-                }
-
-                if (response.data.entry && response.data.entry[0]) {
-                    callback(null, new root.Pivot(svc, response.data.entry[0].content));
-                }
-                else {
-                    callback(new Error("Didn't get a Pivot report back from Splunk"), response);
-                }
-            });
+            return svc.get(Paths.pivot + "/" + encodeURIComponent(this.dataModelObject.dataModel.name), args, response_timeout).
+                        then((response) => {
+                            if (response.data.entry && response.data.entry[0]) {
+                                return new root.Pivot(svc, response.data.entry[0].content);
+                            }
+                            else {
+                                return [new Error("Didn't get a Pivot report back from Splunk"), response];
+                            }
+                        }).catch((err) => {
+                            // TODO: analyze use of response argument and change accordingly
```

Comment:
I wouldn't worry about this too much, this would be a new scenario that we weren't handling previously - omitting the response seems perfectly fine to me